--- libafdt-0.1.0/Makefile.am~	2009-10-24 02:37:25.000000000 +0200
+++ libafdt-0.1.0/Makefile.am	2011-01-22 15:46:04.976237572 +0100
@@ -17,6 +17,7 @@
 # Only enable the asynchronous interface if libevent is available.
 if HAVE_LIBEVENT
 libafdt_la_SOURCES += src/async.c
+libafdt_la_LIBADD = $(LIBEVENT_LIBS)
 endif
 
 include_HEADERS = \
@@ -28,7 +29,7 @@
 
 noinst_PROGRAMS = catter sync_catter
 PROGRAMS_CPPFLAGS = $(LIBEVENT_CPPFLAGS) -I $(top_srcdir)/src
-PROGRAMS_LDADD = libafdt.la $(LIBEVENT_LIBS)
+PROGRAMS_LDADD = libafdt.la
 
 catter_SOURCES = test/catter.c
 catter_CPPFLAGS = $(PROGRAMS_CPPFLAGS)
